### PR TITLE
Added support for project scoped feeds

### PR DESCRIPTION
**Pull request by aasim**

### CODE REVIEW

File: Tasks/DotNetCoreCLIV2/pushcommand.ts [hunk 1]
```diff
@@ -99,8 +99,16 @@ export async function run(): Promise<void> {
                 tempNuGetPath,
                 false /* useNugetToModifyConfigFile */);
 
-            const internalFeedId = tl.getInput('feedPublish');
-            feedUri = await nutil.getNuGetFeedRegistryUrl(packagingLocation.DefaultPackagingUri, internalFeedId, null, accessToken, /* useSession */ true);
+            const feedProject = tl.getInput('feedPublish');
+            var project = null;
+            var internalFeedId = feedProject;
+            if(feedProject && feedProject.includes("/")) {
```
Comment (sachinma) [inline Tasks/DotNetCoreCLIV2/pushcommand.ts]: Should this be a common method across two dot net core cli task commands

Comment (aasim) [reply Tasks/DotNetCoreCLIV2/pushcommand.ts]: Fixed

File: Tasks/DownloadPackageV1/main.ts [hunk 2]
```diff
@@ -35,6 +35,14 @@ async function main(): Promise<void> {
             return Promise.resolve();
         }
 
+        var feedId = projectFeed;
+        var project = null;
+        if(projectFeed.includes("/")) {
```
Comment (jotaylo) [inline Tasks/DownloadPackageV1/main.ts]: any reason not to use getProjectIdAndFeedIdFromInputParam here?

File: Tasks/DownloadPackageV1/task.json [hunk 3]
```diff
@@ -122,15 +121,15 @@
...
+            "resultTemplate": "{ \"Value\" : \"{{#if project}}{{{project.id}}}{{/if}}\\/{{{id}}}\", \"DisplayValue\" : \"{{{name}}}\" }"
         },
         {
             "target": "view",
             "endpointId": "tfs:feed",
             "parameters": {
                 "feed": "$(feed)"
             },
-            "endpointUrl": "{{endpoint.url}}/_apis/packaging/feeds/{{{feed}}}/views",
+            "endpointUrl": "{{endpoint.url}}/{{ #regex ([a-fA-F0-9\\-]*\/)[a-fA-F0-9\\-]+ feed }}_apis/packaging/feeds/{{ #regex [a-fA-F0-9\\-]*\/([a-fA-F0-9\\-]+) feed }}/views",
```
Comment (jotaylo) [inline Tasks/DownloadPackageV1/task.json]: this seriously works? huh.

File: Tasks/DownloadPackageV0/task.loc.json [hunk 4]
```diff
@@ -63,7 +67,7 @@
       "endpointId": "tfs:feed",
       "endpointUrl": "{{endpoint.url}}/_apis/packaging/feeds",
       "resultSelector": "jsonpath:$.value[*]",
```
Comment (jotaylo) [inline Tasks/DownloadPackageV0/task.loc.json]: I wonder if project scoped feeds can be excluded in the jsonpath query instead

File: Tasks/DownloadPackageV1/main.ts (hunk 2 above)
Comment (aasim) [reply Tasks/DownloadPackageV1/main.ts]: Fixed

File: Tasks/DownloadPackageV0/task.loc.json (hunk 4 above)
Comment (aasim) [reply Tasks/DownloadPackageV0/task.loc.json]: I tried using value[?(@project == null)], however it doesn't work. It works the other way around value[?(@project != null)], but not the way we want it to.

File: Tasks/NuGetRestoreV1/package.json [hunk 5]
```diff
@@ -17,10 +17,10 @@
   },
   "homepage": "https://github.com/Microsoft/azure-pipelines-tasks#readme",
   "dependencies": {
+    "azure-pipelines-task-lib": "2.8.0",
```
Comment (jotaylo) [inline Tasks/NuGetRestoreV1/package.json]: >azure-pipelines-task-lib [](start = 5, length = 24)

thanks for doing this one, I must have missed it

File: Tasks/NuGetRestoreV1/nugetinstaller.ts [hunk 6]
```diff
@@ -135,14 +136,19 @@ async function main(): Promise<void> {
...
-                sources.push(<IPackageSource>
-                {
-                    feedName: feed,
-                    feedUri: feedUrl
-                })
+            let feed = getProjectAndFeedIdFromInputParam("feed");
+            
+            if (feed.feedId) {
+                if(feed.projectId) {
+                    tl.warning(tl.loc("UnsupportedProjectScopedFeeds"));
```
Comment (jotaylo) [inline Tasks/NuGetRestoreV1/nugetinstaller.ts]: >warning [](start = 23, length = 7)

This should be a failure, not just a warning.

Comment (aasim) [reply Tasks/NuGetRestoreV1/nugetinstaller.ts]: Fixed

**Pull request merged**